src/Stratis.Bitcoin.IntegrationTests/SmartContracts/SmartContractWalletTests.cs

```diff
@@ -195,7 +195,7 @@ public void SendAndReceiveSmartContractTransactions()
                 // Ensure that boths nodes has the contract
                 senderState = scSender.FullNode.NodeService<ContractStateRepositoryRoot>();
                 receiverState = scReceiver.FullNode.NodeService<ContractStateRepositoryRoot>();
-                tokenContractAddress = transferContractTransaction.GetNewContractAddress();
+                tokenContractAddress = transferContractTransaction.GetNewContractAddress(0); // nonce is 0 for user contract creation.
```

Comment:
@codingupastorm I agree with @majikandy on this one. Can you remove the comments here and move to the summary of the method? I.e. describe that the nonce is used for. I can just see the devs using this going forward will be definitely be interested in what the use of the nonce is here.